projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -394,35 +394,88 @@ function wpcom_launchpad_track_completed_task( $task_id, $extra_props = array()
 }
 
 /**
- * Mark a task as complete.
+ * Returns a list of mappings from 'id_map' to the actual task id.
  *
- * @param string $task_id The task ID.
- * @return bool True if the task was marked as complete, false otherwise.
+ * @return array
  */
-function wpcom_mark_launchpad_task_complete( $task_id ) {
+function wpcom_launchpad_get_reverse_id_mappings() {
 	$task_definitions = wpcom_launchpad_get_task_definitions();
 
-	// If the task ID isn't defined, return false.
-	if ( ! isset( $task_definitions[ $task_id ] ) ) {
-		return false;
+	$mapping = array();
+	foreach ( $task_definitions as $task_id => $value ) {
+		if ( ! isset( $value['id_map'] ) ) {
+			continue;
+		}
+		$mapping[ $value['id_map'] ] = $task_id;
 	}
 
-	// If the task has an id_map, use that instead.
-	$key = $task_id;
-	if ( isset( $task_definitions[ $task_id ]['id_map'] ) ) {
-		$key = $task_definitions[ $task_id ]['id_map'];
-	}
+	return $mapping;
+}
+/**
+ * Mark a task as complete.
+ *
+ * @param string|string[] $task_ids The task IDs that should be marked as complete.
+ * @return bool True if the task was marked as complete, false otherwise.
+ */
+function wpcom_mark_launchpad_task_complete( $task_ids ) {
+	$task_ids = is_array( $task_ids ) ? $task_ids : array( $task_ids );
+	$values   = array_fill_keys( $task_ids, true );
 
-	$statuses         = get_option( 'launchpad_checklist_tasks_statuses', array() );
-	$statuses[ $key ] = true;
-	$result           = update_option( 'launchpad_checklist_tasks_statuses', $statuses );
+	$result = wpcom_update_launchpad_task_status( $values );
 
 	// Record the completion event in Tracks.
-	wpcom_launchpad_track_completed_task( $key );
+	foreach ( $task_ids as $key => $value ) {
+		wpcom_launchpad_track_completed_task( $key );
+	}
```

Comment:
The result of the `array_fill_keys()` call is being put into `$values`, so I think we should be looping over the values in `$task_ids`.
```suggestion
	foreach ( $task_ids as $task_id ) {
		wpcom_launchpad_track_completed_task( $task_id );
	}
```

I also have a concern that we're performing validation of the tasks in `wpcom_update_launchpad_task_status()`, which means we might log the completion here despite the task not being processed. Can we move this logic to `wpcom_update_launchpad_task_status()` so we track completion when it is actually done?